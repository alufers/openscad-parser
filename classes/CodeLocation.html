<!DOCTYPE html><html class="default"><head><meta charSet="utf-8"/><meta http-equiv="x-ua-compatible" content="IE=edge"/><title>CodeLocation | openscad-parser</title><meta name="description" content="Documentation for openscad-parser"/><meta name="viewport" content="width=device-width, initial-scale=1"/><link rel="stylesheet" href="../assets/style.css"/><link rel="stylesheet" href="../assets/highlight.css"/><script async src="../assets/search.js" id="search-script"></script></head><body><script>document.body.classList.add(localStorage.getItem("tsd-theme") || "os")</script><header><div class="tsd-page-toolbar"><div class="container"><div class="table-wrap"><div class="table-cell" id="tsd-search" data-base=".."><div class="field"><label for="tsd-search-field" class="tsd-widget search no-caption">Search</label><input type="text" id="tsd-search-field"/></div><ul class="results"><li class="state loading">Preparing search index...</li><li class="state failure">The search index is not available</li></ul><a href="../index.html" class="title">openscad-parser</a></div><div class="table-cell" id="tsd-widgets"><div id="tsd-filter"><a href="#" class="tsd-widget options no-caption" data-toggle="options">Options</a><div class="tsd-filter-group"><div class="tsd-select" id="tsd-filter-visibility"><span class="tsd-select-label">All</span><ul class="tsd-select-list"><li data-value="public">Public</li><li data-value="protected">Public/Protected</li><li data-value="private" class="selected">All</li></ul></div> <input type="checkbox" id="tsd-filter-inherited" checked/><label class="tsd-widget" for="tsd-filter-inherited">Inherited</label><input type="checkbox" id="tsd-filter-externals" checked/><label class="tsd-widget" for="tsd-filter-externals">Externals</label></div></div><a href="#" class="tsd-widget menu no-caption" data-toggle="menu">Menu</a></div></div></div></div><div class="tsd-page-title"><div class="container"><ul class="tsd-breadcrumb"><li><a href="../modules.html">openscad-parser</a></li><li><a href="CodeLocation.html">CodeLocation</a></li></ul><h1>Class CodeLocation </h1></div></div></header><div class="container container-main"><div class="row"><div class="col-8 col-content"><section class="tsd-panel tsd-hierarchy"><h3>Hierarchy</h3><ul class="tsd-hierarchy"><li><span class="target">CodeLocation</span></li></ul></section><section class="tsd-panel-group tsd-index-group"><h2>Index</h2><section class="tsd-panel tsd-index-panel"><div class="tsd-index-content"><section class="tsd-index-section "><h3>Constructors</h3><ul class="tsd-index-list"><li class="tsd-kind-constructor tsd-parent-kind-class"><a href="CodeLocation.html#constructor" class="tsd-kind-icon">constructor</a></li></ul></section><section class="tsd-index-section "><h3>Properties</h3><ul class="tsd-index-list"><li class="tsd-kind-property tsd-parent-kind-class"><a href="CodeLocation.html#char" class="tsd-kind-icon">char</a></li><li class="tsd-kind-property tsd-parent-kind-class"><a href="CodeLocation.html#col" class="tsd-kind-icon">col</a></li><li class="tsd-kind-property tsd-parent-kind-class"><a href="CodeLocation.html#file" class="tsd-kind-icon">file</a></li><li class="tsd-kind-property tsd-parent-kind-class"><a href="CodeLocation.html#line" class="tsd-kind-icon">line</a></li></ul></section><section class="tsd-index-section tsd-is-private tsd-is-private-protected"><h3>Accessors</h3><ul class="tsd-index-list"><li class="tsd-kind-get-signature tsd-parent-kind-class tsd-is-private"><a href="CodeLocation.html#filename" class="tsd-kind-icon">filename</a></li></ul></section><section class="tsd-index-section "><h3>Methods</h3><ul class="tsd-index-list"><li class="tsd-kind-method tsd-parent-kind-class"><a href="CodeLocation.html#formatWithContext" class="tsd-kind-icon">format<wbr/>With<wbr/>Context</a></li><li class="tsd-kind-method tsd-parent-kind-class"><a href="CodeLocation.html#toString" class="tsd-kind-icon">to<wbr/>String</a></li></ul></section></div></section></section><section class="tsd-panel-group tsd-member-group "><h2>Constructors</h2><section class="tsd-panel tsd-member tsd-kind-constructor tsd-parent-kind-class"><a id="constructor" class="tsd-anchor"></a><h3 class="tsd-anchor-link">constructor<a href="#constructor" aria-label="Permalink" class="tsd-anchor-icon"><svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-link" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round"><path stroke="none" d="M0 0h24v24H0z" fill="none"></path><path d="M10 14a3.5 3.5 0 0 0 5 0l4 -4a3.5 3.5 0 0 0 -5 -5l-.5 .5"></path><path d="M14 10a3.5 3.5 0 0 0 -5 0l-4 4a3.5 3.5 0 0 0 5 5l.5 -.5"></path></svg></a></h3><ul class="tsd-signatures tsd-kind-constructor tsd-parent-kind-class"><li class="tsd-signature tsd-kind-icon">new <wbr/>Code<wbr/>Location<span class="tsd-signature-symbol">(</span>file<span class="tsd-signature-symbol">?: </span><span class="tsd-signature-type">null</span><span class="tsd-signature-symbol"> | </span><a href="CodeFile.html" class="tsd-signature-type" data-tsd-kind="Class">CodeFile</a>, char<span class="tsd-signature-symbol">?: </span><span class="tsd-signature-type">number</span>, line<span class="tsd-signature-symbol">?: </span><span class="tsd-signature-type">number</span>, col<span class="tsd-signature-symbol">?: </span><span class="tsd-signature-type">number</span><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><a href="CodeLocation.html" class="tsd-signature-type" data-tsd-kind="Class">CodeLocation</a></li></ul><ul class="tsd-descriptions"><li class="tsd-description"><aside class="tsd-sources"><ul><li>Defined in <a href="https://github.com/alufers/openscad-parser/blob/1c12113/src/CodeLocation.ts#L9">src/CodeLocation.ts:9</a></li></ul></aside><h4 class="tsd-parameters-title">Parameters</h4><ul class="tsd-parameters"><li><h5>file: <span class="tsd-signature-type">null</span><span class="tsd-signature-symbol"> | </span><a href="CodeFile.html" class="tsd-signature-type" data-tsd-kind="Class">CodeFile</a><span class="tsd-signature-symbol"> = null</span></h5></li><li><h5>char: <span class="tsd-signature-type">number</span><span class="tsd-signature-symbol"> = 0</span></h5></li><li><h5>line: <span class="tsd-signature-type">number</span><span class="tsd-signature-symbol"> = 0</span></h5></li><li><h5>col: <span class="tsd-signature-type">number</span><span class="tsd-signature-symbol"> = 0</span></h5></li></ul><h4 class="tsd-returns-title">Returns <a href="CodeLocation.html" class="tsd-signature-type" data-tsd-kind="Class">CodeLocation</a></h4></li></ul></section></section><section class="tsd-panel-group tsd-member-group "><h2>Properties</h2><section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class"><a id="char" class="tsd-anchor"></a><h3 class="tsd-anchor-link"><span class="tsd-flag ts-flagReadonly">Readonly</span> char<a href="#char" aria-label="Permalink" class="tsd-anchor-icon"><svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-link" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round"><path stroke="none" d="M0 0h24v24H0z" fill="none"></path><path d="M10 14a3.5 3.5 0 0 0 5 0l4 -4a3.5 3.5 0 0 0 -5 -5l-.5 .5"></path><path d="M14 10a3.5 3.5 0 0 0 -5 0l-4 4a3.5 3.5 0 0 0 5 5l.5 -.5"></path></svg></a></h3><div class="tsd-signature tsd-kind-icon">char<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">number</span><span class="tsd-signature-symbol"> = 0</span></div><aside class="tsd-sources"><ul><li>Defined in <a href="https://github.com/alufers/openscad-parser/blob/1c12113/src/CodeLocation.ts#L29">src/CodeLocation.ts:29</a></li></ul></aside><div class="tsd-comment tsd-typography"><div class="lead">
<p>The character offset in the file contents.</p>
</div></div></section><section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class"><a id="col" class="tsd-anchor"></a><h3 class="tsd-anchor-link"><span class="tsd-flag ts-flagReadonly">Readonly</span> col<a href="#col" aria-label="Permalink" class="tsd-anchor-icon"><svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-link" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round"><path stroke="none" d="M0 0h24v24H0z" fill="none"></path><path d="M10 14a3.5 3.5 0 0 0 5 0l4 -4a3.5 3.5 0 0 0 -5 -5l-.5 .5"></path><path d="M14 10a3.5 3.5 0 0 0 -5 0l-4 4a3.5 3.5 0 0 0 5 5l.5 -.5"></path></svg></a></h3><div class="tsd-signature tsd-kind-icon">col<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">number</span><span class="tsd-signature-symbol"> = 0</span></div><aside class="tsd-sources"><ul><li>Defined in <a href="https://github.com/alufers/openscad-parser/blob/1c12113/src/CodeLocation.ts#L39">src/CodeLocation.ts:39</a></li></ul></aside><div class="tsd-comment tsd-typography"><div class="lead">
<p>The column number of this location. Zero-indexed.</p>
</div></div></section><section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class"><a id="file" class="tsd-anchor"></a><h3 class="tsd-anchor-link"><span class="tsd-flag ts-flagReadonly">Readonly</span> file<a href="#file" aria-label="Permalink" class="tsd-anchor-icon"><svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-link" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round"><path stroke="none" d="M0 0h24v24H0z" fill="none"></path><path d="M10 14a3.5 3.5 0 0 0 5 0l4 -4a3.5 3.5 0 0 0 -5 -5l-.5 .5"></path><path d="M14 10a3.5 3.5 0 0 0 -5 0l-4 4a3.5 3.5 0 0 0 5 5l.5 -.5"></path></svg></a></h3><div class="tsd-signature tsd-kind-icon">file<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">null</span><span class="tsd-signature-symbol"> | </span><a href="CodeFile.html" class="tsd-signature-type" data-tsd-kind="Class">CodeFile</a></div><aside class="tsd-sources"><ul><li>Defined in <a href="https://github.com/alufers/openscad-parser/blob/1c12113/src/CodeLocation.ts#L24">src/CodeLocation.ts:24</a></li></ul></aside><div class="tsd-comment tsd-typography"><div class="lead">
<p>THe file to which this location points.</p>
</div></div></section><section class="tsd-panel tsd-member tsd-kind-property tsd-parent-kind-class"><a id="line" class="tsd-anchor"></a><h3 class="tsd-anchor-link"><span class="tsd-flag ts-flagReadonly">Readonly</span> line<a href="#line" aria-label="Permalink" class="tsd-anchor-icon"><svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-link" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round"><path stroke="none" d="M0 0h24v24H0z" fill="none"></path><path d="M10 14a3.5 3.5 0 0 0 5 0l4 -4a3.5 3.5 0 0 0 -5 -5l-.5 .5"></path><path d="M14 10a3.5 3.5 0 0 0 -5 0l-4 4a3.5 3.5 0 0 0 5 5l.5 -.5"></path></svg></a></h3><div class="tsd-signature tsd-kind-icon">line<span class="tsd-signature-symbol">:</span> <span class="tsd-signature-type">number</span><span class="tsd-signature-symbol"> = 0</span></div><aside class="tsd-sources"><ul><li>Defined in <a href="https://github.com/alufers/openscad-parser/blob/1c12113/src/CodeLocation.ts#L34">src/CodeLocation.ts:34</a></li></ul></aside><div class="tsd-comment tsd-typography"><div class="lead">
<p>The line number of this location. Zero-indexed.</p>
</div></div></section></section><section class="tsd-panel-group tsd-member-group tsd-is-private tsd-is-private-protected"><h2>Accessors</h2><section class="tsd-panel tsd-member tsd-kind-get-signature tsd-parent-kind-class tsd-is-private"><a id="filename" class="tsd-anchor"></a><h3 class="tsd-anchor-link"><span class="tsd-flag ts-flagPrivate">Private</span> filename<a href="#filename" aria-label="Permalink" class="tsd-anchor-icon"><svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-link" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round"><path stroke="none" d="M0 0h24v24H0z" fill="none"></path><path d="M10 14a3.5 3.5 0 0 0 5 0l4 -4a3.5 3.5 0 0 0 -5 -5l-.5 .5"></path><path d="M14 10a3.5 3.5 0 0 0 -5 0l-4 4a3.5 3.5 0 0 0 5 5l.5 -.5"></path></svg></a></h3><ul class="tsd-signatures tsd-kind-get-signature tsd-parent-kind-class tsd-is-private"><li class="tsd-signature tsd-kind-icon"><span class="tsd-signature-symbol">get</span> filename<span class="tsd-signature-symbol">(</span><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">string</span></li></ul><ul class="tsd-descriptions"><li class="tsd-description"><aside class="tsd-sources"><ul><li>Defined in <a href="https://github.com/alufers/openscad-parser/blob/1c12113/src/CodeLocation.ts#L70">src/CodeLocation.ts:70</a></li></ul></aside><h4 class="tsd-returns-title">Returns <span class="tsd-signature-type">string</span></h4></li></ul></section></section><section class="tsd-panel-group tsd-member-group "><h2>Methods</h2><section class="tsd-panel tsd-member tsd-kind-method tsd-parent-kind-class"><a id="formatWithContext" class="tsd-anchor"></a><h3 class="tsd-anchor-link">format<wbr/>With<wbr/>Context<a href="#formatWithContext" aria-label="Permalink" class="tsd-anchor-icon"><svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-link" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round"><path stroke="none" d="M0 0h24v24H0z" fill="none"></path><path d="M10 14a3.5 3.5 0 0 0 5 0l4 -4a3.5 3.5 0 0 0 -5 -5l-.5 .5"></path><path d="M14 10a3.5 3.5 0 0 0 -5 0l-4 4a3.5 3.5 0 0 0 5 5l.5 -.5"></path></svg></a></h3><ul class="tsd-signatures tsd-kind-method tsd-parent-kind-class"><li class="tsd-signature tsd-kind-icon">format<wbr/>With<wbr/>Context<span class="tsd-signature-symbol">(</span><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">string</span></li></ul><ul class="tsd-descriptions"><li class="tsd-description"><aside class="tsd-sources"><ul><li>Defined in <a href="https://github.com/alufers/openscad-parser/blob/1c12113/src/CodeLocation.ts#L47">src/CodeLocation.ts:47</a></li></ul></aside><h4 class="tsd-returns-title">Returns <span class="tsd-signature-type">string</span></h4></li></ul></section><section class="tsd-panel tsd-member tsd-kind-method tsd-parent-kind-class"><a id="toString" class="tsd-anchor"></a><h3 class="tsd-anchor-link">to<wbr/>String<a href="#toString" aria-label="Permalink" class="tsd-anchor-icon"><svg xmlns="http://www.w3.org/2000/svg" class="icon icon-tabler icon-tabler-link" viewBox="0 0 24 24" stroke-width="2" stroke="currentColor" fill="none" stroke-linecap="round" stroke-linejoin="round"><path stroke="none" d="M0 0h24v24H0z" fill="none"></path><path d="M10 14a3.5 3.5 0 0 0 5 0l4 -4a3.5 3.5 0 0 0 -5 -5l-.5 .5"></path><path d="M14 10a3.5 3.5 0 0 0 -5 0l-4 4a3.5 3.5 0 0 0 5 5l.5 -.5"></path></svg></a></h3><ul class="tsd-signatures tsd-kind-method tsd-parent-kind-class"><li class="tsd-signature tsd-kind-icon">to<wbr/>String<span class="tsd-signature-symbol">(</span><span class="tsd-signature-symbol">)</span><span class="tsd-signature-symbol">: </span><span class="tsd-signature-type">string</span></li></ul><ul class="tsd-descriptions"><li class="tsd-description"><aside class="tsd-sources"><ul><li>Defined in <a href="https://github.com/alufers/openscad-parser/blob/1c12113/src/CodeLocation.ts#L41">src/CodeLocation.ts:41</a></li></ul></aside><h4 class="tsd-returns-title">Returns <span class="tsd-signature-type">string</span></h4></li></ul></section></section></div><div class="col-4 col-menu menu-sticky-wrap menu-highlight"><nav class="tsd-navigation primary"><ul><li class=""><a href="../modules.html">Exports</a></li></ul></nav><nav class="tsd-navigation secondary menu-sticky"><ul><li class="current tsd-kind-class"><a href="CodeLocation.html" class="tsd-kind-icon">Code<wbr/>Location</a><ul><li class="tsd-kind-constructor tsd-parent-kind-class"><a href="CodeLocation.html#constructor" class="tsd-kind-icon">constructor</a></li><li class="tsd-kind-property tsd-parent-kind-class"><a href="CodeLocation.html#char" class="tsd-kind-icon">char</a></li><li class="tsd-kind-property tsd-parent-kind-class"><a href="CodeLocation.html#col" class="tsd-kind-icon">col</a></li><li class="tsd-kind-property tsd-parent-kind-class"><a href="CodeLocation.html#file" class="tsd-kind-icon">file</a></li><li class="tsd-kind-property tsd-parent-kind-class"><a href="CodeLocation.html#line" class="tsd-kind-icon">line</a></li><li class="tsd-kind-get-signature tsd-parent-kind-class tsd-is-private"><a href="CodeLocation.html#filename" class="tsd-kind-icon">filename</a></li><li class="tsd-kind-method tsd-parent-kind-class"><a href="CodeLocation.html#formatWithContext" class="tsd-kind-icon">format<wbr/>With<wbr/>Context</a></li><li class="tsd-kind-method tsd-parent-kind-class"><a href="CodeLocation.html#toString" class="tsd-kind-icon">to<wbr/>String</a></li></ul></li></ul></nav></div></div></div><footer class="with-border-bottom"><div class="container"><h2>Legend</h2><div class="tsd-legend-group"><ul class="tsd-legend"><li class="tsd-kind-constructor tsd-parent-kind-interface"><span class="tsd-kind-icon">Constructor</span></li><li class="tsd-kind-property tsd-parent-kind-interface"><span class="tsd-kind-icon">Property</span></li><li class="tsd-kind-method tsd-parent-kind-interface"><span class="tsd-kind-icon">Method</span></li></ul><ul class="tsd-legend"><li class="tsd-kind-constructor tsd-parent-kind-class"><span class="tsd-kind-icon">Constructor</span></li><li class="tsd-kind-property tsd-parent-kind-class"><span class="tsd-kind-icon">Property</span></li><li class="tsd-kind-method tsd-parent-kind-class"><span class="tsd-kind-icon">Method</span></li></ul><ul class="tsd-legend"><li class="tsd-kind-constructor tsd-parent-kind-class tsd-is-inherited"><span class="tsd-kind-icon">Inherited constructor</span></li><li class="tsd-kind-property tsd-parent-kind-class tsd-is-inherited"><span class="tsd-kind-icon">Inherited property</span></li><li class="tsd-kind-method tsd-parent-kind-class tsd-is-inherited"><span class="tsd-kind-icon">Inherited method</span></li></ul><ul class="tsd-legend"><li class="tsd-kind-property tsd-parent-kind-class tsd-is-protected"><span class="tsd-kind-icon">Protected property</span></li><li class="tsd-kind-method tsd-parent-kind-class tsd-is-protected"><span class="tsd-kind-icon">Protected method</span></li></ul><ul class="tsd-legend"><li class="tsd-kind-property tsd-parent-kind-class tsd-is-private"><span class="tsd-kind-icon">Private property</span></li><li class="tsd-kind-method tsd-parent-kind-class tsd-is-private"><span class="tsd-kind-icon">Private method</span></li></ul><ul class="tsd-legend"><li class="tsd-kind-property tsd-parent-kind-class tsd-is-static"><span class="tsd-kind-icon">Static property</span></li><li class="tsd-kind-method tsd-parent-kind-class tsd-is-static"><span class="tsd-kind-icon">Static method</span></li></ul></div><h2>Settings</h2><p>Theme <select id="theme"><option value="os">OS</option><option value="light">Light</option><option value="dark">Dark</option></select></p></div></footer><div class="container tsd-generator"><p>Generated using <a href="https://typedoc.org/" target="_blank">TypeDoc</a></p></div><div class="overlay"></div><script src="../assets/main.js"></script></body></html>