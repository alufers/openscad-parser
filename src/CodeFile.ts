import * as fs from "fs";
import * as path from "path";

export default class CodeFile {
  constructor(public path: string, public code: string) {}

  get filename() {
    return path.basename(this.path);
  }

  /**
   * Loads an openscad file from the filesystem.
   */
  static async load(pathToLoad: string): Promise<CodeFile> {
    pathToLoad = path.resolve(pathToLoad); // normalize the path
    const contents = await new Promise<string>((res, rej) => {
      fs.readFile(
        pathToLoad,
        {
          encoding: "utf8",
        },
        (err, data) => {
          if (err) {
            rej(err);
            return;
          }
          res(data);
        }
      );
    });
    return new CodeFile(pathToLoad, contents);
  }
}
