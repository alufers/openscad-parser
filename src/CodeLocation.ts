import CodeFile from "./CodeFile";

export default class CodeLocation {
  constructor(
    file: CodeFile = null,
    char: number = 0,
    line: number = 0,
    col: number = 0
  ) {
    this.file = file;
    this.char = char;
    this.line = line;
    this.col = col;
  }

  /**
   * THe file to which this location points.
   */
  file: CodeFile;

  /**
   * The character offset in the fil;e contents.
   */
  char = 0;

  /**
   * The line number of this location. Zero-indexed.
   */
  line = 0;

  /**
   * The column number of this location. Zero-indexed.
   */
  col = 0;

  toString(): string {
    return `file '${this.file.filename}' line ${this.line + 1} column ${this
      .col + 1}'`;
  }
}
