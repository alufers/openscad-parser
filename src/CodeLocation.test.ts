import CodeLocation from "./CodeLocation";
import CodeFile from "./CodeFile";

describe("CodeLocation", () => {
  it("stringifies itself and matches snapshot", () => {
    const str = new CodeLocation(new CodeFile("s", "d"), 34, 4, 8).toString();
    expect(str).toMatchSnapshot();
  });
  it("constructs", () => {
    new CodeLocation();
  });
});
