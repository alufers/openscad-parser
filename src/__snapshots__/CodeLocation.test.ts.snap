// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeLocation stringifies itself and matches snapshot 1`] = `"file 's' line 5 column 9'"`;
